astropy/table/index.py

```diff
@@ -825,6 +841,51 @@ def __getitem__(self, item):
         return self.table[rows]
 
 
+class TableLocIndices(TableLoc):
+    def __init__(self, table):
+        super().__init__(table)
+
+    def __getitem__(self, item):
+        '''
+                Retrieve Table row's indices by value slice.
+
+                Parameters
+                ----------
+                item : column element, list, ndarray, slice or tuple
+                    Can be a value of the table primary index, a list/ndarray
+                    of such values, or a value slice (both endpoints are included).
+                    If a tuple is provided, the first element must be
+                    an index to use instead of the primary key, and the
+                    second element must be as above.
+                '''
+        return self._TableLoc__get_row(item)
+
+    def __setitem__(self, key, value):
```

Comment:
Forgot this: `__setitem__` should go in the `TableLoc` class instead, as it's goal is to allow updating directly the rows.